### PR TITLE
redux

### PR DESCRIPTION
**Pull request by GCerbelli**
## Summary

- Refactor booklist adding redux

## Trello Card

[Nº 5.1 Redux I - Actions, Reducers, Store](https://trello.com/c/sDUSV2U7/25-n-51-redux-i-actions-reducers-store)

## Screenshots
image

### CODE REVIEW

File: wbooks/src/app/screens/BooksList/index.tsx [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
-import { Book } from '../../interfaces/books';
+import { Book, BooksState } from '../../interfaces/books';
 import Routes from '../../../constants/routes';
+import { actionCreator } from '../../../redux/books/actions';
 
 import styles from './styles';
 
 const BooksList = ({ navigation }: any) => {
+  const dispatch = useDispatch();
+  const books = useSelector((state: BooksState) => state.books.books, shallowEqual);
```
Comment (marianozicavo) [inline wbooks/src/app/screens/BooksList/index.tsx:14]:
```suggestion
  const books = useSelector((state: AppState) => state.books.books, shallowEqual);
```

File: wbooks/src/redux/books/reducer.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  switch (action.type) {
+    case actions.GET_BOOKS:
+      return {
+        ...state,
+        loading: true
+      };
+    case actions.GET_BOOKS_SUCCESS:
+      return {
+        ...state,
+        loading: false,
```
Comment (guidoprinc) [inline wbooks/src/redux/books/reducer.js:19]: booksLoading

File: wbooks/src/redux/books/reducer.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  booksLoading: false,
+  booksError: ''
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case actions.GET_BOOKS:
+      return {
+        ...state,
+        loading: true
```
Comment (guidoprinc) [inline wbooks/src/redux/books/reducer.js:14]: booksloading

File: wbooks/src/redux/books/reducer.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    case actions.GET_BOOKS_SUCCESS:
+      return {
+        ...state,
+        loading: false,
+        books: action.payload
+      };
+    case actions.GET_BOOKS_FAILURE:
+      return {
+        ...state,
+        loading: false,
```
Comment (guidoprinc) [inline wbooks/src/redux/books/reducer.js:25]: booksloading

File: wbooks/src/redux/books/reducer.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        error: 'Error'
```
Comment (guidoprinc) [inline wbooks/src/redux/books/reducer.js:26]: bookserror

File: wbooks/src/redux/books/reducer.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  booksLoading: false,
+  booksError: ''
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case actions.GET_BOOKS:
+      return {
+        ...state,
+        booksLoading: true
```
Comment (guidoprinc) [inline wbooks/src/redux/books/reducer.js:14]: you should clear the error too when you are doing the action at the beginning

**Pull request merged**